Lib/python/pyrun.swg

```diff
@@ -394,18 +409,23 @@ SwigPyObject_repr(SwigPyObject *v)
 {
   const char *name = SWIG_TypePrettyName(v->ty);
   PyObject *repr = SWIG_Python_str_FromFormat("<Swig Object of type '%s' at %p>", (name ? name : "unknown"), (void *)v);
-  if (v->next) {
+  if (repr && v->next) {
```

Comment:
Further OOM fixes:

If there is an allocation failure, `SwigPyObject_repr` will return NULL, otherwise it will return a reference to the string containing the full representation.  All object references to intermediate strings allocated during building the representation will be freed, in both cases.

Firstly, handle OOM on building the string for the current SWIG object, to avoid calling `PyUnicode_Concat` or `PyString_ConcatAndDel` below with `repr = NULL`.